worker/task_executor.go

```diff
@@ -0,0 +1,164 @@
+package worker
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/docker/docker/api/types/mount"
+	"github.com/docker/docker/client"
+	"github.com/ufcg-lsd/arrebol-pb-worker/utils"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+)
+
+const (
+	TaskScriptExecutorFileName  = "task-script-executor.sh"
+	RunTaskScriptCommandPattern = "/bin/bash %s -d -tsf=%s"
+	DefaultWorkerDockerImage    = "ubuntu"
+)
+
+type TaskExecutor struct {
+	Cli client.Client
+	Cid string
+}
+
+func (e *TaskExecutor) Execute(task *Task, containerSpawnWarner chan<- interface{}) error {
+	image := task.Image
+
+	if image == "" {
+		image = DefaultWorkerDockerImage
+	}
+	log.Println("Creating container with image: " + image)
+
+	config := utils.ContainerConfig{
+		Name:   "",
+		Image:  image,
+		Mounts: []mount.Mount{},
+	}
+
+	if err := e.initiate(config); err != nil {
+		return err
+	}
+	if err := e.send(task); err != nil {
+		return err
+	}
+	// It allows the system to know when the container
+	// is ready.
+	containerSpawnWarner <- "spawned"
```

Comment:
the idea is to start a goroutine to call the Execute function?

Reply:
actually there is a goroutine to report the task status, and it needs to know when the container is ready to be inspected. In the next pr it's gonna be clearer.

Reply:
if we are no using goroutines, why do we need a channel here?

Reply:
we are, the report goroutine reads from this channel to know that the container is ready to be inspected. But this code is coming with the next pr

Reply:
ok